### DISCUSSION TITLE
Where am I supposed to put shared logic by components (or pages)

### DISCUSSION THREAD

**Discussion by manuelaguirre** [nuxt/nuxt#18300] [Questions]
Let's say I have a component that performs a UseFetch with special logic for the treatment of the data that is fetched. Where should I put the logic that I extract from the `<script>` in order to import them from other components?

**Comment by wobsoriano** [comment]
That's actually up to you. You can have a `utils` page for example in your root folder and import it like

```html
<script setup lang="ts">
import { sayHello } from '~/utils/say-hello'
</script>

<template>
  <div>
    {{ sayHello('robert') }}
  </div>
</template>
```

For composables, a `composables` folder is recommended 

```ts
// composables/useSomeExpensiveCalculation.ts
import type { Ref } from 'vue'
import type { Post } from '~/types/post'

export const useSomeExpensiveCalculation = (result: Ref<Post>) => {
    const result = computed(() => someFunction(result.value))
    return result
}
```

```html
<script setup lang="ts">
import { useSomeExpensiveCalculation } from '~/composables/useSomeExpensiveCalculation';

const { data } = await useFetch('/api/post')

const result = useSomeExpensiveCalculation(data)
</script>
```

Also worth noting that they're working on auto-import of composables https://github.com/nuxt/framework/pull/1176

**Reply by manuelaguirre** [reply]
Thank you so much. Are these "modules" very different from composables? or do they solve entirely different problems?

**Reply by wobsoriano** [reply]
Nuxt modules extend your app configurations, plugins, etc.. they are called sequentially when booting Nuxt. https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#modules
